howso/input_validation.amlg

```diff
@@ -441,113 +441,173 @@
-										(~ (null) given_value)
+										(if (!~ (null) given_value)
+											[{ message "Value must be null."}]
+
+											[]
+										)
 
 										(= exp_type "number")
-										(and
-											(~ 0 given_value)
-											(if (contains_index specification "max")
-												(<= given_value (get specification "max"))
-												(true)
+										(if (!~ 0 given_value)
+											[{ message "Value must be a number."}]
+
+											(and
+												(contains_index specification "max")
+												(> given_value (get specification "max"))
 											)
-											(if (contains_index specification "min")
-												(>= given_value (get specification "min"))
-												(true)
+											[{ message (concat "Value must less than or equal to " (get specification "max") ".")}]
+
+											(and
+												(contains_index specification "min")
+												(< given_value (get specification "min"))
 											)
-											(if (contains_index specification "exclusive_max")
-												(< given_value (get specification "exclusive_max"))
-												(true)
+											[{ message (concat "Value must greater than or equal to " (get specification "min") ".")}]
+
+											(and
+												(contains_index specification "exclusive_max")
+												(>= given_value (get specification "exclusive_max"))
 											)
-											(if (contains_index specification "exclusive_min")
-												(> given_value (get specification "exclusive_min"))
-												(true)
+											[{ message (concat "Value must less than " (get specification "exclusive_max") ".")}]
+
+											(and
+												(contains_index specification "exclusive_min")
+												(<= given_value (get specification "exclusive_min"))
 											)
+											[{ message (concat "Value must greater than " (get specification "exclusive_min") ".")}]
+
+											[]
 										)
 
 										(= exp_type "string")
-										(if (contains_index specification "enum")
-											(contains_value
-												(get specification "enum")
-												given_value
+										(if (!~ "" given_value)
+											[{ message "The given value must be a string."}]
+
+											(and
+												(contains_index specification "enum")
+												(not (contains_value (get specification "enum") given_value))
 											)
+											[{
+												message
+													(concat
+														"The given value: " given_value ", is not one of "
+														(apply "concat" (trunc (weave
+															(get specification "enum")
+															", "
+														)))
+														"."
+													)
+											}]
 
-											;else just check if string
-											(~ "" given_value)
+											[]
 										)
 
 										(= exp_type "boolean")
-										(contains_value (list (true) (false)) given_value)
+										(if (contains_value (list (true) (false)) given_value)
+											[]
+											[{ message "The given value must be a boolean value." }]
```

Comment:
Looks like there are different variations for the same message for each type. Lets make them consistent.